openmetadata-ui/src/main/resources/ui/src/components/AdvancedSearch/AdvancedSearch.interface.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ *  Copyright 2021 Collate
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import { JsonTree } from 'react-awesome-query-builder';
+import { SearchIndex } from '../../enums/search.enum';
+
+export interface AdvancedSearchProps {
+  jsonTree: JsonTree | undefined;
+  searchIndex: SearchIndex;
+  onChangeJsonTree: (tree: JsonTree) => void;
+  onChangeQueryFilter: (
+    queryFilter: Record<string, unknown> | undefined
+  ) => void;
+}
+
+export type FilterObject = Record<string, string[]>;
+
+export function isFilterObject(obj: unknown): obj is FilterObject {
+  const typedObj = obj as FilterObject;
+
+  return (
+    ((typedObj !== null && typeof typedObj === 'object') ||
+      typeof typedObj === 'function') &&
```

Comment:
This was generated using [ts-type-guard](https://github.com/rhys-vdw/ts-auto-guard).

I am not 100% why its checking that `typeof typedObj === 'function'`, but passing a function to `Object.entries` returns `[]`, so that may be the reason

```js
const f = (arg) => arg + 1;
-> undefined
typeof f
-> 'function'
Object.entries(f)
-> []
```